src/java/org/apache/cassandra/service/snapshot/SnapshotManager.java

```diff
@@ -116,8 +108,15 @@ public synchronized void addSnapshot(TableSnapshot snapshot)
     @VisibleForTesting
     protected synchronized void loadSnapshots()
     {
-        logger.debug("Loading snapshots");
-        snapshotLoader.get().forEach(this::addSnapshot);
+        SnapshotLoader loader = new SnapshotLoader(DatabaseDescriptor.getAllDataFileLocations());
```

Comment:
Dont you think we might have `loader` as a property of `SnapshotManager` ? Why do we need to create an instance every time?

btw, I just went through that SnapshotLoader class as such and I would change one of the constructors to this:

    public SnapshotLoader(String[] dataDirectories)
    {
        this(Arrays.stream(dataDirectories).map(Paths::get).collect(Collectors.toList()));
    }

I know it is not relevant to this patch and changes related to this but I find it just easy to change, really. Just an idea.